src/dune_rules/coq/coq_lib.ml

```diff
@@ -244,32 +387,46 @@ module DB = struct
       let+ () =
         if allow_private_deps then Resolve.Memo.return ()
         else
-          match theory.package with
-          | Some _ -> Resolve.Memo.return ()
-          | None -> Error.private_deps_not_allowed ~loc theory_name
+          match theory with
+          | Dune { package = None; _ } ->
+            Error.private_deps_not_allowed ~loc theory_name
+          | Legacy _ | Dune _ -> Resolve.Memo.return ()
       in
       (loc, theory)
 
     let resolve_theories ~coq_lang_version ~allow_private_deps ~coq_db ~boot_id
-        theories =
+        ~loc ~use_stdlib ~is_boot theories =
+      (* If the user has not provided Coq as a dependency and (use_stdlib yes)
+         is set, we add it here. This will be used in extraction and dune coq
+         top. *)
+      let theories =
+        if
+          use_stdlib && (not is_boot)
+          && not
+             @@ List.exists theories ~f:(fun (_, name) ->
+                    Coq_lib_name.equal Coq_lib_name.stdlib name)
+        then (loc, Coq_lib_name.stdlib) :: theories
+        else theories
+      in
       let f =
         resolve_theory ~coq_lang_version ~allow_private_deps ~coq_db ~boot_id
       in
       Resolve.Memo.List.map theories ~f
 
     let create_from_stanza_impl (coq_db, db, dir, (s : Coq_stanza.Theory.t)) =
-      let name = snd s.name in
-      let id = Id.create ~path:dir ~name:s.name in
+      let name = s.name in
+      let id = Id.create ~path:(Path.build dir) ~name in
       let coq_lang_version = s.buildable.coq_lang_version in
       let open Memo.O in
       let boot_id = if s.boot then None else boot_library_id coq_db in
       let allow_private_deps = Option.is_none s.package in
       let use_stdlib = s.buildable.use_stdlib in
       let+ libraries =
-        resolve_plugins ~db ~allow_private_deps ~name s.buildable.plugins
+        resolve_plugins ~db ~allow_private_deps ~name:(snd name)
+          s.buildable.plugins
       and+ theories =
         resolve_theories ~coq_lang_version ~allow_private_deps ~coq_db ~boot_id
-          s.buildable.theories
+          ~loc:s.buildable.loc ~use_stdlib ~is_boot:s.boot s.buildable.theories
       and+ boot = resolve_boot ~coq_lang_version ~coq_db boot_id in
       let theories =
```

Comment:
Moreover note that this is done outside resolve_theories as at some point will become guarded by a version flag, requiring users to add the Coq theory explicilty.